### ISSUE TITLE
Create an image based on Postgres 9.5

### ISSUE THREAD

**Issue by md5** [appropriate/docker-postgis#8]
cf. https://github.com/docker-library/postgres/pull/70

**Comment by md5** [comment]
Just checked again today whether there's a PostGIS 2.1 package available for Postgres 9.5 and there doesn't appear to be one yet.

**Comment by ImreSamu** [comment]
+1 

Is it  possible to create 9.5-2.2 version ?  ("PostgreSQL 9.5 with PostGIS 2.2 ) 

>"PostgreSQL 9.5 and PostGIS 2.2 are both in development. We plan to not support PostgreSQL 9.5 in 2.1 (changes have already happened in 9.5 that make it incompatible with 2.1 (these fixes have been put in 2.2 but not backported to 2.1))  https://trac.osgeo.org/postgis/wiki/UsersWikiPostgreSQLPostGIS

I am testing programs with different postgre + postgis versions, 
so  Is it possible to create all Postgre + PostGis supported versions ?
like : 
*  9.0-1.5 ; 9.0-2.0 
*  9.1-1.5 ; 9.1-2.0 ; 9.1-2.2
*  9.2-1.5 ; 9.2-2.0 ; 9.2-2.2
*  9.3-2.2
*  9.4-2.2

( https://trac.osgeo.org/postgis/wiki/UsersWikiPostgreSQLPostGIS#PostGISSupportMatrix )

**Comment by md5** [comment]
I just checked again and I don't see any PostGIS 2.2 packages at all (I checked both [`jessie-pgdg`](http://apt.postgresql.org/pub/repos/apt/dists/jessie-pgdg/main/binary-amd64/Packages) and  [`jessie-pgdg-testing`](http://apt.postgresql.org/pub/repos/apt/dists/jessie-pgdg-testing/main/binary-amd64/Packages)). I also don't see any 1.5 or 2.0 packages for any Postgres version.

```console
$ curl -fsSL http://apt.postgresql.org/pub/repos/apt/dists/jessie-pgdg/main/binary-amd64/Packages | grep -w 'Package: postgresql-.*postgis-2.1$' 

Package: postgresql-9.0-postgis-2.1
Package: postgresql-9.1-postgis-2.1
Package: postgresql-9.2-postgis-2.1
Package: postgresql-9.3-postgis-2.1
Package: postgresql-9.4-postgis-2.1
```

If you can figure out where the packages are (if they exist) and provide a PR that adds the relevant directories and updates the `update.sh` script, I'll be willing to merge it.

This isn't the first time that I've ran into issues with the maintenance of the PostGIS Debian packages. The more common problem is that packages that used to exist in the index suddenly disappear when a new version is packaged. This has broken the automated build for this package after a new build was triggered by an update to the `postgres` image more times that I've kept count of.

**Comment by md5** [comment]
I just realized that last `curl` command was a bit off (it's past my bedtime). But even with the right `curl` command there are only 2.1 packages in `jessie-pgdg`:

```console
$ curl -fsSL http://apt.postgresql.org/pub/repos/apt/dists/jessie-pgdg/main/binary-amd64/Packages | grep -w 'Package: postgresql-.*postgis-.*$' |
 grep -v -- -scripts
Package: postgresql-9.0-postgis-2.1
Package: postgresql-9.1-postgis-2.1
Package: postgresql-9.2-postgis-2.1
Package: postgresql-9.3-postgis-2.1
Package: postgresql-9.4-postgis-2.1
```

It's possible that those packages exist under `wheezy-pgdg` or some other distro, but I'm only supporting `jessie` since that's what the base `postgres` image is based on.

**Comment by md5** [comment]
Looks like there are 2.0 packages for `wheezy`, but I can't find any 1.5 packages:

```console
$ curl -fsSL http://apt.postgresql.org/pub/repos/apt/dists/wheezy-pgdg/main/binary-amd64/Packages | grep 'Package:.*-postgis-' | grep -v -- -scripts
Package: postgresql-8.4-postgis-2.0
Package: postgresql-9.0-postgis-2.0
Package: postgresql-9.0-postgis-2.1
Package: postgresql-9.1-postgis-2.0
Package: postgresql-9.1-postgis-2.1
Package: postgresql-9.2-postgis-2.0
Package: postgresql-9.2-postgis-2.1
Package: postgresql-9.3-postgis-2.1
Package: postgresql-9.4-postgis-2.1
$ curl -fsSL http://apt.postgresql.org/pub/repos/apt/dists/squeeze-pgdg/main/binary-amd64/Packages | grep 'Package:.*-postgis-' | grep -v -- -scripts
$ curl -fsSL http://apt.postgresql.org/pub/repos/apt/dists/etch-pgdg/main/binary-amd64/Packages | grep 'Package:.*-postgis-' | grep -v -- -scripts
$ 
```

**Comment by md5** [comment]
FWIW, I've started a branch for 9.5-2.2: https://github.com/appropriate/docker-postgis/tree/postgres-9.5-postgis-2.2

It doesn't yet have a package version in it, since there isn't a package yet, but I'll update it when the first packages are published.

**Comment by emilbayes** [comment]
Looks like [RMPs available for Postgres 9.5](https://lists.osgeo.org/pipermail/postgis-users/2015-October/040854.html), maybe that helps further the work on this? :)

**Comment by md5** [comment]
@emilbayes Thanks for the heads-up, but it looks like only RPMs were pushed, not `*.deb` packages. Using RPMs would involve a lot of work switching the base image to `centos` and would be a breaking change for existing users.

**Comment by ImreSamu** [comment]
I have found 9.4-2.2  in the  debian "experimental" repo :  postgresql-9.4-postgis-2.2 , 
* https://packages.debian.org/experimental/postgresql-9.4-postgis-2.2
* https://packages.debian.org/search?keywords=postgresql-9.4-postgis-2.2

**Comment by md5** [comment]
Thanks @ImreSamu. I saw those packages as well. I haven't had a chance to look at how they differ from the packages provided by the `pgdg` repositories at http://apt.postgresql.org, but I'd be hesitant to switch to them unless they're fully compatible.

**Comment by ImreSamu** [comment]
> I'd be hesitant to switch to them unless they're fully compatible.
ok , no problem ...  :)

**Comment by willfarrell** [comment]
9.5 is now stable and used as the latest at https://hub.docker.com/_/postgres/

Should be a lot easier to add it in now.

**Comment by hildo** [comment]
Postgres 9.5 and PostGis 2.2 would be good.  Happy to lend a hand if it will help move things along.

**Comment by md5** [comment]
I still don't see packages in the Apt repo for PostGIS 2.2. I also don't see PostGIS 2.1 packages for Postgres 9.5, so I'm afraid we're stuck with the packages we have until that situation is resolved.

If someone can point me to where I can get the packages or provide a PR, I'm happy to review.

```console
$ curl -fsSL http://apt.postgresql.org/pub/repos/apt/dists/jessie-pgdg/main/binary-amd64/Packages | grep -w 'Package: postgresql-.*postgis-.*$' |  grep -v -- -script
s
Package: postgresql-9.0-postgis-2.1
Package: postgresql-9.1-postgis-2.1
Package: postgresql-9.2-postgis-2.1
Package: postgresql-9.3-postgis-2.1
Package: postgresql-9.4-postgis-2.1
```

**Comment by md5** [comment]
Looks like @sebastic has been working on the Debian package for PostGIS 2.2 over at https://anonscm.debian.org/cgit/pkg-grass/postgis.git

Hope you don't mind the ping @sebastic, but do you have a rough sense of when a PostGIS 2.2 package for Debian will be available and whether it will be targeting `jessie` or `squeeze`? It looks like Postgres 9.5 has been packaged for `jessie` by the PGDG (cf. https://github.com/docker-library/postgres/commit/2c72165e4f9dc66a61998111f69b5eba0b6b71f4), so it would be awesome if I could update this Docker image to PostGIS 2.2 and still keep it based on the [`postgres`](https://github.com/docker-library/postgres) image.

**Comment by md5** [comment]
Sorry, I meant `stretch`, not `squeeze`.

**Comment by sebastic** [comment]
PostGIS 2.2.1 was uploaded to Debian unstable yesterday. Markus Wanner takes care of the postgis package in PGDG/pgApt, this is complicated somewhat because SFCGAL is only available in stretch & sid.

When PostGIS 2.2.1 migrates to testing I'll update it along with sfcgal for jessie-backports.

**Comment by md5** [comment]
Thanks @sebastic!

**Comment by ImreSamu** [comment]
:)

```bash
$ curl -fsSL http://apt.postgresql.org/pub/repos/apt/dists/jessie-pgdg/main/binary-amd64/Packages | grep -w 'Package: postgresql-.*postgis-.*$' |  grep -v -- -scripts
Package: postgresql-9.0-postgis-2.1
Package: postgresql-9.1-postgis-2.1
Package: postgresql-9.1-postgis-2.2
Package: postgresql-9.2-postgis-2.1
Package: postgresql-9.2-postgis-2.2
Package: postgresql-9.3-postgis-2.1
Package: postgresql-9.3-postgis-2.2
Package: postgresql-9.4-postgis-2.1
Package: postgresql-9.4-postgis-2.2
Package: postgresql-9.5-postgis-2.2
```

**Comment by md5** [comment]
Thanks for the bump @ImreSamu. Travis checks are running on a PR now: https://github.com/appropriate/docker-postgis/pull/17

**Comment by md5** [comment]
Unfortunately, it looks like the updates made to `apt.postgresql.org` included removing the 2.0 and 2.1 versions of the `postgis` package :sob: 

@sebastic Do you know who manages `apt.postgresql.org`? Disappearing package versions have been a recurring problem with that repository.

**Comment by sebastic** [comment]
To quote the [Apt page on wiki.postgresql.org](https://wiki.postgresql.org/wiki/Apt) to which you're redirected from apt.postgresql.org:
> The PostgreSQL Global Development Group (PGDG) maintains an APT repository of PostgreSQL packages for Debian and Ubuntu located at http://apt.postgresql.org/pub/repos/apt/. We aim at building PostgreSQL server packages as well as extensions and modules packages on several Debian/Ubuntu releases for all PostgreSQL versions supported. 

That page further documents:
>  **Contact**
>  
> * Mailing list: pgsql-pkg-debian@postgresql.org ([Archives](http://archives.postgresql.org/pgsql-pkg-debian/))
> * IRC channel: #postgresql-apt @ irc.freenode.net
> * People: Christoph Berg <myon@debian.org>, Dimitri Fontaine <dimitri@2ndquadrant.fr>, Magnus Hagander <magnus@hagander.net> 
>  
> **Bugs**
>  
> Please report bugs on the pgsql-pkg-debian@postgresql.org mailing list, or open a bug in the [Debian BTS](http://bugs.debian.org/).

**Comment by md5** [comment]
Thanks @sebastic!

**Comment by md5** [comment]
Looks like that `postgis` package only includes the PostGIS userland binaries:

```
/usr/bin/pgsql2shp
/usr/bin/shp2pgsql
/usr/bin/raster2pgsql
/usr/bin/shp2pgsql-gui
```

I believe I can update the 2.1 Docker images to stop installing `postgis`, but then those binaries will no longer be available.

**Comment by md5** [comment]
I just sent a message to `pgsql-pkg-debian@postgresql.org`. I realize now that providing both 2.1 and 2.2 versions of `postgis` in the same repository would be problematic since the packages don't have version numbers in their names.

I'm wondering now whether the 2.2 version of the PostGIS userland binaries are still compatible with an installation of `postgresql-9.x-postgis-2.1`. If so, I can just unpin the version of the `postgis` package to get the images to build.

**Comment by md5** [comment]
I got a very helpful response from @mwanner on `pgsql-pkg-debian` and the conclusion was that this image should only support PostGIS 2.2 going forward. Since Postgres 9.0 is EOL, I've opened an issue in the `docker-library/postgres` repository to remove the `postgres:9.0` image from the list of supported tags (see https://github.com/docker-library/postgres/issues/121).

By removing support for Postgres 9.0, we can limit this image to only those Postgres releases that are not EOL, all of which have published packages from PostGIS 2.2.

I'll update this PR later today with the fixes.
